install_config/persistent_storage/persistent-volume-snapshots.adoc

```diff
@@ -0,0 +1,401 @@
+metadata:
+  name: volumesnapsot-admin
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: volumesnapshot-admin
+subjects:
+- kind: User
+  name: alice
+----
+
+[NOTE]
+====
+This is only an example of API access configuration. The `VolumeSnapshot` objects behave similar to other [{product-title}] API objects. See the link:https://kubernetes.io/docs/admin/accessing-the-api/[API access control documentation] for more information on managing the API RBAC.
+====
+
+[[lifecycle]]
+== Lifecycle of a Volume Snapshot and Volume Snapshot Data
+
+[[pvc-and-pv]]
+=== Persistent Volume Claim and Persistent Volume
+
+The `PersistentVolumeClaim` is bound to a `PersistentVolume`. The `PersistentVolume` type must be one of the snapshot supported persistent volume types.
+
+[[snapshot-promoter]]
+==== Snapshot Promoter
+
+To create a `StorageClass`:
+
+[source,yaml]
+----
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: snapshot-promoter
+provisioner: volumesnapshot.external-storage.k8s.io/snapshot-promoter
+----
+
+This `StorageClass` is necessary to restore a `PersistentVolume` from a `VolumeSnapshot` that was previously created.
+
+[[create-snapshot]]
+=== Create Snapshot
+
+To take a snapshot of a PV, create a new `VolumeSnapshot` object:
+
+[source,yaml]
+----
+apiVersion: volumesnapshot.external-storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: snapshot-demo
+spec:
+  persistentVolumeClaimName: ebs-pvc
+----
+
+`persistentVolumeClaimName` is the name of the `PersistentVolumeClaim` bound to
+a `PersistentVolume`. This particular PV is snapshotted.
+
+A `VolumeSnapshotData` object is then automatically created based on the
+`VolumeSnapshot`. The relationship between `VolumeSnapshot` and
+`VolumeSnapshotData` is similar to the relationship between
+`PersistentVolumeClaim` and `PersistentVolume`.
+
+Depending on the PV type, the operation might go through several phases, which
+are reflected by the `VolumeSnapshot` status:
+
+. The new `VolumeSnapshot` object is created.
+. The controller starts the snapshot operation. The snapshotted `PersistentVolume` might need to be frozen and the applications paused.
+. The storage system finishes creating the snapshot (the snapshot is "cut") and the snapshotted `PersistentVolume` might return to normal operation. The snapshot itself is not yet ready. The last status condition is of `Pending` type with status value `True`. A new `VolumeSnapshotData` object is created to represent the actual snapshot.
+. The newly created snapshot is complete and ready to use. The last status condition is of `Ready` type with status value `True`.
+
+[IMPORTANT]
+====
+It is the user's responsibility to ensure data consistency (stop the pod/application, flush caches, freeze the file system, and so on). 
+====
+
+[NOTE]
+====
+In case of error, the `VolumeSnapshot` status is appended with an `Error` condition.
+====
+
+To display the `VolumeSnapshot` status:
+
+----
+sh
```

Comment:
Is this `sh` supposed to be the codeblock `source` highlighting type? Or an actual command.

Reply:
Deleted.